index.html

```diff
@@ -4596,78 +4596,80 @@ <h2>Key and Signature Expiration</h2>
-        using the old cryptographic material can now instead be generated using the new
-        material and can be verified using the new verification method.
-      </p>
-
-      <p>
-        Rotation is a useful mechanism for protecting against verification
-        method compromise, since frequent rotation of a verification method by
-        the controller reduces the value of a single compromised verification method
-        to an attacker. Performing revocation immediately after rotation is
-        useful for verification methods that a controller designates for
-        short-lived verifications, such as those involved in encrypting messages
-        and authentication.
-      </p>
-
-      <p class="note">
-        Higher security environments tend to employ more frequent verification
-        method rotation.
-      </p>
-
-      <p class="note">
-        Frequent rotation of a verification method might be frustrating for
-        parties that are forced to continuously renew or refresh associated
-        credentials.
-      </p>
+      <ul>
+        <li>
+<a>Verification method</a> rotation is a proactive security measure.
+        </li>
+        <li>
+It is considered a best practice to perform <a>verification method</a> rotation
+on a regular basis.
+        </li>
+        <li>
+Higher security environments tend to employ more frequent verification method
+rotation.
+        </li>
+        <li>
+<a>Verification method</a> rotation applies only to the current or latest
+version of a <a>DID document</a>.
+        </li>
+        <li>
+When a <a>verification method</a> has been active for a long time, or used for
+many operations, a controller might wish to perform a rotation.
+        </li>
+        <li>
+Frequent rotation of a <a>verification method</a> might be frustrating for
+parties that are forced to continuously renew or refresh associated credentials.
+        </li>
+        <li>
+Proofs or signatures that rely on <a>verification methods</a> that are not
+present in the latest version of a <a>DID document</a> are not impacted by
+rotation. In these cases, verification software might require additional
+information, such as when a particular <a>verification method</a> was
+expected to be valid as well as access to a <a>verifiable data registry</a>
+containing a historical record, to determine the validity of the proof or
+signature.
+        </li>
+        <li>
+The section on <a href="#method-operations">DID method operations</a> specifies
+the <a>DID</a> operations to be supported by a <a>DID method</a> specification,
+including <a href="#method-operations">update</a> which is expected to be used
+to perform a <a>verification method</a> rotation.
+        </li>
+        <li>
+A <a>controller</a> performs a rotation when they add a new <a>verification
+method</a> that is meant to replace an existing <a>verification method</a> after
+some time.
```

Comment:
Is rotation meant to communicate an intention to replace exactly at the moment of the rotation -- or at some indefinite point in the future? If the former interpretation is canonical, then isn't revocation of the old method automatic?

Reply:
My expectation is that it's "at some indefinite point in the future"... which might be within microseconds of each other. The process, as I understand it might be:

1. Add replacement key (k2).
2. Remove old key (k1).

So, it can't be considered automatic all of the time (but perhaps, sometimes). Yes, the future moment could be thought of as when the old key is revoked.

@dhh1128 please take a shot at some concrete spec text, or @OR13 if you have any suggestions here, that would be great. We need that text before I can merge.

Reply:
Under the definition you're advocating here, the only difference between "inserting" a new key and "rotating" a key is intention; there is no observable difference in DID doc state when the two operations complete. This feels weird to me. I thought "rotation" was the combination of adding a new key and removing an old one, as an atomic operation.

Reply:
I'm not suggesting spec text here because I think the definition is fundamentally wrong. If this definition is maintained, I think the whole section needs to be rewritten. For example, the earlier text "enables the secret cryptographic material associated with an existing verification method to be deactivated or destroyed once a new verification method; has been added to the DIDdocument" has to be revised, since it implies the opposite order (that rotation happens *after* revocation).

Reply:
I don't have input on this PR. I was merely cleaning up text that was already there when I stepped on this landmine. 

I believe other people submitted this text initially and I would like them to work through the details here. I remember it being @dhh1128 and @OR13 working on the initial PR here, but can't remember exactly.

I don't necessarily care about the outcome here as long as it's consistent. I do think @dhh1128 is raising good points, so I expect that anything that makes it in would address those concerns. To avoid a "too many cooks in the kitchen" scenario, I'm recusing myself from providing input.

If it helps, I can try to hunt down who initially worked on and refined this section of the specification.

Reply:
Did a quick lookup to save you the time @msporny 

The original text was merged via PR #569 with an alternative originally proposed by @OR13 in #548. This was originally intended to address issue #386 

It looks like @dhh1128 and @OR13 were able to come to agreement after some back and forth on the editing of this. Hopefully the discussion between those two will provide some additional context on what allowed the current text to be satisfying to all parties at that time.